✨[Feat] : id가 없을 때의 예외처리

Yejin-Moon opened the pull request:
### Pull Request 타입(하나 이상의 PR 타입을 선택해주세요)
- [x] 기능 추가
- [ ] 기능 삭제
- [ ] 버그 수정
- [ ] 의존성, 환경 변수, 빌드 관련 코드 업데이트

### 반영 브랜치
SPOFO-143 -> main

### 변경 사항
DB에 찾으려는 social ID가 없을 때 IllegalArgumentException를 발생시켜서 예외를 처리했습니다.

### 테스트 결과
헤더에서 받은 ID 값을 DB에서 찾을 수 없다면 errorCode 와 errorMessage를 반환합니다.

Yejin-Moon committed:
✨[Feat] : id가 없을 때의 예외처리

DB에 찾으려는 social ID가 없을 때 IllegalArgumentException를 발생시켜서 예외를 처리했습니다.

JANG-WON-JUN commented on src/main/java/net/spofo/auth/controller/MemberController.java:
```diff
@@ -20,8 +21,12 @@ public class MemberController {
         String searchedMember = httpHeaders.getFirst("authorization");
-        Long id = memberRepository.findBySocialId(searchedMember).getId();
-        return id;
+        Member m = memberRepository.findBySocialId(searchedMember);
```
1. 먼저 findBySocialId 메서드는 서비스를 통해 호출하는 것이 좋을 것 같습니다. (레이어 지키키)
2. 그리고 예외를 throw하는 주체는 서비스가 되면 좋을 것 같습니다.

JANG-WON-JUN commented on src/main/java/net/spofo/auth/exception/controller/ExceptionController.java:
```diff
@@ -0,0 +1,29 @@
+
+    @ResponseBody
+    @ExceptionHandler
+    public ResponseEntity<ErrorResult> invalidRequestHandler(IllegalArgumentException e) {
```
IllegalArgumentException 를 직접 받는 것도 좋지만 Auth 서버의 최상위 exception을 만들어서 상속받아 커스텀 예외를 만드는게 더 확장성에 좋을 것 같으니 검토 해 보시면 좋을 것 같습니다.

JANG-WON-JUN commented on src/main/java/net/spofo/auth/exception/dto/ErrorResult.java:
```diff
@@ -0,0 +1,19 @@
+    private final String errorMessage;
+
+    @Builder
+    public ErrorResult(HttpStatus errorCode, String errorMessage) {
```
@Builder를 쓰는 이유는 생성자를 대신하기 위함임니다.
생성자의 접근 제한자를 public이 아닌 private로 해도 Builder는 작동할거예요.
또 lombok의 builder 쪽 로직을 확인하시면 더 공부가 될 것 같아요.

Yejin-Moon committed:
🔨[Refactor] : 코드 리팩토링

findBySocialId 메소드를 서비스에서 호출
NoSocialIdException 추가

SeungGyu-Park reviewed (approved):
확인했습니다.

JANG-WON-JUN commented on src/main/java/net/spofo/auth/controller/MemberController.java:
```diff
@@ -14,14 +15,13 @@
         String searchedMember = httpHeaders.getFirst("authorization");
-        Long id = memberRepository.findBySocialId(searchedMember).getId();
-        return id;
+        Member member = memberService.findBySocialId(searchedMember);
```
searchedMember의 변수명을 다시 한 번 확인 해주시면 좋을 것 같아요

JANG-WON-JUN commented on src/main/java/net/spofo/auth/controller/MemberController.java:
```diff
@@ -14,14 +15,13 @@
-        Long id = memberRepository.findBySocialId(searchedMember).getId();
-        return id;
+        Member member = memberService.findBySocialId(searchedMember);
+        return member.getId();
```
Contoller에서는 ResponseEntity를 사용하여 응답하면 좋을 것 같아요

JANG-WON-JUN commented on src/main/java/net/spofo/auth/service/MemberService.java:
```diff
@@ -28,4 +30,13 @@ public String getPortfolio() {
     }
+
+    public Member findBySocialId(String socialId) {
+        Member member = memberRepository.findBySocialId(socialId);
```
MemberRepository의 findBySocialId의 반환값을 Optional로 바꾸면 member가 존재하지 않을 때의 예외처리를 더 간단하게 작성할 수 있습니다.

Yejin-Moon committed:
🔨[Refactor] : 코드 리팩토링

변수명(searchedMember -> searchedId) 변경
컨트롤러에서 DTO로 반환하도록 변경
findBySocialId의 반환값을 optional로 변경

JANG-WON-JUN reviewed (approved):
확인했습니다

Yejin-Moon committed:
🔨[Refactor] : 코드 리팩토링

RequestHeader 변경
MemberResponse와 RestClient 생성자 변경
